be/src/vec/aggregate_functions/aggregate_function_map.h

```diff
@@ -135,13 +135,44 @@ struct AggregateFunctionMapAggData {
                                                        num_rows);
         dst_key_column.get_nested_column().insert_range_from(*_key_column, 0, num_rows);
         dst.get_values().insert_range_from(*_value_column, 0, num_rows);
-        if (offsets.size() == 0) {
+        if (offsets.empty()) {
             offsets.push_back(num_rows);
         } else {
             offsets.push_back(offsets.back() + num_rows);
         }
     }
 
+    void write(BufferWritable& buf) const {
+        const size_t size = _key_column->size();
+        write_binary(size, buf);
+        for (size_t i = 0; i < size; i++) {
+            write_binary(assert_cast<KeyColumnType&>(*_key_column).get_data_at(i), buf);
+        }
+        for (size_t i = 0; i < size; i++) {
+            write_binary(_value_column->get_data_at(i), buf);
+        }
+    }
+
+    void read(BufferReadable& buf) {
+        size_t size = 0;
+        read_binary(size, buf);
+        std::string keys;
+        for (size_t i = 0; i < size; i++) {
+            read_binary(keys, buf);
+            StringRef key {keys};
+            if (_map.find(key) != _map.cend()) {
```

Comment:
when will the key not be found?

Reply:
do nothing， the map aggregation function assumed that keys cannot be duplicated, so if there are duplicates, they should not be placed in the map.